src/routes/(auth)/register/+page.server.ts

```diff
@@ -29,24 +33,32 @@ export const actions: Actions = {
 			errors.push('All fields are required');
 		}
 
-		if (email && !emailRegex.test(email?.toString())) {
-			errors.push('Invalid email address');
+		try {
+			if (email) validateEmail(email);
+		} catch (e: any) {
+			errors.push(e.message);
 		}
 
-		if (password && password.toString().length < 8) {
-			errors.push('Password must be at least 8 characters long');
+		try {
+			if (password) validatePassword(password);
```

Comment:
Since the password needs to be validated in both the register and reset-password routes, I moved the password validation logic to a separate module to avoid code duplication, along with functions to validate the email, username, and name.

Reply:
Thanks! The code quality is really good.

Reply:
Thanks for reviewing!